tests/services/application.service.spec.ts

```diff
@@ -1,18 +1,37 @@
 import { ApplicationService } from '../../src/app/core/services/application.service';
 
-const applicationService = new ApplicationService(null);
+const upToDateAppService = new ApplicationService(null);
 const currentVersion = '3.2.5';
 
+class ApplicationServiceStub extends ApplicationService {
+  readonly currentVersion: string;
+  constructor(latestVersion: string, currentVersion: string) {
+    super(null);
+    this.latestVersion = latestVersion;
+    this.currentVersion = currentVersion;
+  }
+}
+
 describe('Test the ApplicationService', () => {
-    it('Test whether the ApplicationService can detect outdated versions', () => {
-        expect(applicationService.isOutdatedVersion('2.2.5', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.1', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.4', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.4.8.9', currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion(currentVersion, currentVersion)).toBe(false);
-        expect(applicationService.isOutdatedVersion('3.2.5.0', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('3.2.5.5', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('3.2.6', currentVersion)).toBe(true);
-        expect(applicationService.isOutdatedVersion('10', currentVersion)).toBe(true);
-    });
+  it('Test whether the ApplicationService can detect outdated versions', () => {
```

Comment:
Maybe we can have the test name such that it includes the `.isApplicationOutdated()` method in the test name. 

For example: `isApplicationOutdated() correctly detects  outdated versions`